packages/amplify-category-api/src/commands/api/add-graphql-datasource.js

```diff
@@ -106,15 +108,18 @@ module.exports = {
         const apiDirPath = `${projectBackendDirPath}/${category}/${resourceName}`;
         fs.ensureDirSync(apiDirPath);
         const graphqlSchemaFilePath = `${apiDirPath}/schema.graphql`;
-        fs.ensureFileSync(graphqlSchemaFilePath);
-        const graphqlSchemaRaw = fs.readFileSync(graphqlSchemaFilePath, 'utf8');
-        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
-
+        const graphqlSchemaRaw = readSchema(apiDirPath);
+        const schemaFileExists = fs.existsSync(graphqlSchemaFilePath);
         const rdsGraphQLSchemaDoc = graphqlSchemaContext.schemaDoc;
-
-        const concatGraphQLSchemaDoc = mergeTypes([currGraphQLSchemaDoc, rdsGraphQLSchemaDoc], { all: true });
-
-        fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
+        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
+        if (schemaFileExists) {
+          const concatGraphQLSchemaDoc = mergeTypes([currGraphQLSchemaDoc, rdsGraphQLSchemaDoc], { all: true });
+          fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
+        } else {
+          const graphqlSchemaFilePath = `${apiDirPath}/rds.graphql`;
+          const concatGraphQLSchemaDoc = mergeTypes([{}, rdsGraphQLSchemaDoc], { all: true });
+          fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
+        }
```

Comment:
```suggestion
        const schemaFileExists = fs.existsSync(graphqlSchemaFilePath);
const rdsGraphQLSchemaDoc = graphqlSchemaContext.schemaDoc;
const schemaDirectoryPath = path.join(projectDirectory, 'schema');

if (schemaFileExists) {
    const graphqlSchemaRaw = readSchema(apiDirPath);
    const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
    const concatGraphQLSchemaDoc = mergeTypes([currGraphQLSchemaDoc, rdsGraphQLSchemaDoc], { all: true });
    fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
} else  {
    fs.ensureDirSync(schemaDirectoryPath);
    const graphqlSchemaFilePath = path.join('apiDirPath', 'rds.graphql');
    const concatGraphQLSchemaDoc = mergeTypes([{}, rdsGraphQLSchemaDoc], { all: true });
    fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
}
```

Reply:
I have removed the readSchema function since it was only reading schema file